### PR TITLE
Add Depth First Search in Java

### LINKED ISSUE #3342: Add Depth First Search in Java
To request a new code snippet, please fill out the following:

Project name: Depth First Search
Project link: https://sampleprograms.io/projects/depth-first-search
Language: Java

> The project link must be a link to a project on the [Sample Programs Website project list][sample-programs-project-list].
> If you would like to add a new project to the Sample Programs Website,
> please first make a pull request to the [Sample Programs Website][sample-programs-website].

[sample-programs-website]: https://github.com/TheRenegadeCoder/sample-programs-website
[sample-programs-project-list]: https://sampleprograms.io/projects/

### PR DESCRIPTION
**Pull request by koishi-daisuki**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][renegade-coder]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

Please fill _one_ of the sections below as applicable.
Please also add any other relevant information to the Notes section at the bottom.
You may delete or just ignore any other sections.
For more information please refer to our [contributing documentation][contributing]

## I Am Adding a New Code Snippet in an Existing Language

- [x] I fixed #3342 
- [x] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format

## Other Notes

Implement the Depth First Search project in Java

[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-new-language]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-language
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sampleprograms.io/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#modifying-existing-code-snippets

### CODE REVIEW

File: archive/j/java/DepthFirstSearch.java
Comment (rzuckerm) [inline archive/j/java/DepthFirstSearch.java:39]: This character (`。`) needs to be removed from the string in order to get the tests to pass with Java 23

File: archive/j/java/DepthFirstSearch.java
Comment (rzuckerm) [inline archive/j/java/DepthFirstSearch.java:20]: Unfortunately, we are running Java 11 which does not seem to support the `toList` method. However, I tested your code and all of the current code with Java 23, and everything seemed to pass (with the exception of the error message having an invalid character). Please edit `archive/j/java/testinfo.yml` and change `11-slim` to `23-slim`.

**Review by rzuckerm — APPROVED** [review]
Hi @koishi-daisuki, thanks for your submission, but there are a few things that need to be fix before I can merge this PR. Also, is this intended for Hacktoberfest? If so, I can wait until October 1 to merge this so that you can get credit.

**Comment by koishi-daisuki** [comment]
> Hi @koishi-daisuki, thanks for your submission, but there are a few things that need to be fixed before I can merge this PR. Also, is this intended for Hacktoberfest? If so, I can wait until October 1 to merge this so that you can get credit.

Yes, This is intended for Hacktoberfest! thank you so much

**Comment by koishi-daisuki** [comment]
I also fixed the bugs of invalid character and change "11-slim" to "23-slim"

**Comment by rzuckerm** [comment]
@koishi-daisuki Looks good! I'll wait until October 1 to merge this

**Comment by rzuckerm** [comment]
Actually, it looks like Hacktoberfest is now open, so I'm merging this now.

**Pull request merged**
